web/src/components/jobs/JobTags.tsx

```diff
@@ -0,0 +1,290 @@
+            startIcon={<LocalOfferIcon />}
+          >
+            Edit Tag
+          </Button>
+        </MQTooltip>
+        <Autocomplete
+          multiple
+          disableCloseOnSelect
+          id='dataset-tags'
+          sx={{ flex: 1, width: 'auto' }}
+          limitTags={6}
+          autoHighlight
+          disableClearable
+          disablePortal
+          options={tagData.map((option) => option.name)}
+          value={selectedTags}
+          onChange={handleTagChange}
+          renderTags={(value: string[]) => formatTags(value, tagData)}
+          renderOption={(props, option, { selected }) => (
+            <li {...props}>
+              <Checkbox
+                icon={<CheckBoxOutlineBlankIcon fontSize='small' />}
+                checkedIcon={<CheckBoxIcon fontSize='small' />}
+                style={{ marginRight: 4 }}
+                checked={selected}
+              />
+              <div>
+                <MQText bold>{option}</MQText>
+                <MQText subdued overflowHidden>
+                  {tagData.find((tagItem) => tagItem.name === option)?.description ||
+                    'No Tag Description'}
+                </MQText>
+              </div>
+            </li>
+          )}
+          renderInput={(params) => (
+            <TextField
+              variant={'outlined'}
+              {...params}
+              placeholder={selectedTags.length > 0 ? '' : 'Search Tags'}
+              InputProps={{
+                ...params.InputProps,
+              }}
+              InputLabelProps={{
+                shrink: true,
+              }}
+              size='small'
+            />
+          )}
+        />
+      </Box>
+      <Dialog
+        PaperProps={{
+          sx: { backgroundColor: theme.palette.background.default, backgroundImage: 'none' },
+        }}
+        open={openTagDesc}
+        fullWidth
+        maxWidth='sm'
+        onKeyDown={(event) => {
+          if (event.key === 'Escape') {
+            handleTagDescClose()
+          }
+        }}
+      >
+        <DialogTitle>Select a Tag to change</DialogTitle>
+        <DialogContent>
+          <MQText subheading>Tag</MQText>
+          <Autocomplete
+            options={tagData.map((option) => option.name)}
+            autoSelect
+            freeSolo
+            fullWidth
+            autoFocus
+            forcePopupIcon
+            onChange={handleTagDescChange}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                placeholder={'Search for a Tag...or enter a new one.'}
+                autoFocus
+                margin='dense'
+                id='tag'
+                variant='outlined'
+                InputLabelProps={{
+                  ...params.InputProps,
+                  shrink: false,
+                }}
+              />
+            )}
+          />
+          <MQText subheading bottomMargin>
+            Description
+          </MQText>
```

Comment:
I was thinking we may want to use the `label` prop on these, maybe it would work a bit better with the design system. I could be wrong.